impl/client/src/main/javascript/web/util/formatting.js

```diff
@@ -192,6 +192,30 @@ define("common-ui/util/formatting", ['common-ui/util/timeutil', 'common-ui/util/
           }
 
           if(timezone == 'client') {
+
+              if (value.length == 28) {
+
+                  var currentDate = new Date();
+                  var offset = -currentDate.getTimezoneOffset();
+
+                  // calculation is done in minutes because it is the smaller unit on a timezone definition
+                  // e.g.: "2018-05-20T01:12:23.456-0430" where -04h30m is the timezone setting
+                  var offsetServerMinutes = value.substring(23, 26) * 60;
```

Comment:
The piece you extract is something like `0430`, which represents `4*60 + 30 = 270` minutes. The code you wrote would interpret this as 430 minutes.
Also, you seem to be ignoring offset's +/- sign (and why the `-` on the offset variable above?).

Also, it is "not considered good practice" to let JavaScript implicitly convert from string to number — it's usually best to write something more explicit, such as `Number(text)` or an explicit `+` sign, as in `(+text)`.

Reply:
Actually in the example above I am using `-04` substring (and then transforming it to `-04*60`), which is also wrong because the intention is to use also the minutes part.

Agree with the `Number(text)` suggestion, so I will use the following statement instead:
`var offsetServerMinutes = Number( (value.substring(23, 26) * 60) + value.substring(26, value.length()) );`

Regarding the `-` signal, the reason is that javascript's method getTimezoneOffset() returns the value as negative in accordance to the timezone.
Example: If your time zone is GMT+2, -120 will be returned (more info on https://www.w3schools.com/jsref/jsref_gettimezoneoffset.asp)

Reply:
You're right about the substring capturing `-04` and not `0430`...
I also did not know about the `getTimezoneOffset` returning the negative of what I expected.
About the use of `Number`, note that you're still relying on the implicit conversion when doing `(value.substring(23, 26) * 60)`.